editor/editor_node.cpp

```diff
@@ -3179,6 +3179,16 @@ void EditorNode::_menu_option_confirm(int p_option, bool p_confirmed) {
 		} break;
 		case SET_RENDERER_NAME_SAVE_AND_RESTART: {
 			ProjectSettings::get_singleton()->set("rendering/renderer/rendering_method", renderer_request);
+			if (renderer_request == "mobile" || renderer_request == "gl_compatibility") {
+				// Also change the mobile override if changing to a compatible rendering method.
+				// This prevents visual discrepancies between desktop and mobile platforms.
+				ProjectSettings::get_singleton()->set("rendering/renderer/rendering_method.mobile", renderer_request);
+			} else if (renderer_request == "forward_plus") {
+				// Use the equivalent mobile rendering method. This prevents the rendering method from staying
+				// on its old choice if moving from `gl_compatibility` to `forward_plus`.
+				ProjectSettings::get_singleton()->set("rendering/renderer/rendering_method.mobile", "mobile");
```

Comment:
This will overwrite the setting even if it was intentionally set to forward_plus or gl_compatibility by the user.

I'm not sure the mobile setting should be secretly changed by the engine without the user's knowledge

Reply:
The dropdown options could be changed to show which option would be used on which platform, e.g. "Forward+ (desktop)/Forward Mobile (mobile".

That said, I don't know if you can display a selected text that is different from the option text (so that the name in the top-right corner isn't too ling).

Reply:
Coming from this newly-linked issue: https://github.com/godotengine/godot/pull/72461.

I think the ease of having these three options in a simple, color-coded dropdown is hard to beat.

I quite like your idea of having the dropdown being an 'upper limit,' where mobile maxes-out at Mobile, and web maxes-out at Compatibility. It would give users control and safety.

Alternatively, setting web and mobile to the same value via this dropdown makes complete sense to me. It's a big-hammer (befitting such a prominent button). And I can always split the platforms back to different renderers using the Project Settings, once I've figured out which platforms suit my project.

Future mobile and XR hardware will likely be capable of rendering with Forward+ (at some point), so not placing a soft limit via the menu does provide a measure of future-proofing.

Having said all that, I'd be in favor of either of these options.

Reply:
I discussed this with @BastiaanOlij on RocketChat and we both agree that Calinou's last suggestion is a good idea. 

I think leaving the dropdown option text as is, but using the configurations suggested make sense. From a user perspective, they will be setting the desktop mode, but the mobile mode will tag along in a way that makes sense. 

I see two situations where there may still be confusion:
1. The user has set desktop and mobile to Forward+ in the project settings, then tests out mobile or compatibility. Once they return to Forward+, the project setting is incorrect
2. The user wants to test mobile on desktop and compatibility on mobile (I understand this is a useful configuration for games that ship on both PCVR and mobile VR)

To alleviate those issues, I suggest that the popup menu includes a short warning i.e. "this will modify the rendering_method project settings from their current values to: ..." So users are warned that they are changing multiple settings.

Reply:
> To alleviate those issues, I suggest that the popup menu includes a short warning i.e. "this will modify the rendering_method project settings from their current values to: ..." So users are warned that they are changing multiple settings.

I've updated the PR accordingly (check OP for screenshots) :slightly_smiling_face: